lib/granite/form/model/associations/nested_attributes.rb

```diff
@@ -162,7 +162,7 @@ def self.assign_to_or_mark_for_destruction(object, attributes, allow_destroy)
             end
 
             def self.destroy_flag?(hash)
-              Granite::Form.typecaster(Boolean).call(hash[DESTROY_ATTRIBUTE])
+              Types::Boolean.typecast(hash[DESTROY_ATTRIBUTE])
```

Comment:
The only reason this worked is because `Boolean` typecaster only needed 1 argument. However if it was overridden in application to also require the second parameter, this would crash.